feat(DIA-863): add tap events to the new home view cards

araujobarret opened the pull request:
The type of this PR is: **Feature**

<!-- Bugfix/Feature/Enhancement/Documentation -->

<!-- If applicable, write the Jira ticket number in square brackets e.g. `[CO-434]`
     The Jira integration will turn it into a clickable link for you. -->

This PR resolves [DIA-863]

### Description

Sibling of #535

Adds:
- `TappedCardsGroup` and `TappedCardsChipsGroup` events
- `collectionsCategory` as owner type

### PR Checklist (tick all before merging)

<!-- 💡 This checklist is experimental. #cohesion warmly welcomes any feedback about the list or how it impacts your workflow -->

- [x] If I've added a new file to the tree I've exported it from the common [`index.ts`](https://github.com/artsy/cohesion/blob/main/src/Schema/Events/index.ts)
- [x] I've added comments with examples for any new interfaces and ensured that they're in the docs
- [x] No platform-specific terminology has been used outside of `click` and `tap` (platform is inferred by the DB storing events)


[DIA-863]: https://artsyproduct.atlassian.net/browse/DIA-863?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

dblandin commented on src/Schema/Events/Tap.ts:
````diff
@@ -327,6 +327,50 @@ export interface TappedFairGroup extends TappedEntityGroup {
+ *  }
+ * ```
+ */
+export interface TappedCardsGroup extends TappedEntityGroup {
````
We usually have the entity in the event name singular: so `tappedCardGroup`.

araujobarret replied:
changed it to singular in the latest commits 👍

dblandin commented on src/Schema/Events/Tap.ts:
````diff
@@ -327,6 +327,50 @@ export interface TappedFairGroup extends TappedEntityGroup {
+ *  }
+ * ```
+ */
+export interface TappedCardsChipsGroup extends TappedEntityGroup {
````
Still not sold on needing both these events. Maybe just start with `tappedCardGroup`? We'll be able to use the contextModule value to differentiate between specific section instances.

araujobarret replied:
makes sense, removed in the last commits 👍

dblandin reviewed (commented):
Looking good. Left a couple comments but let's see what @louislecluse thinks.

louislecluse reviewed (approved):
@daytavares We're introducing "tappedCardGroup" here (for a new component type closer to a 'card'), I think it works with your tracking guideline but let me know otherwise.

dblandin commented on src/Schema/Events/Tap.ts:
```diff
@@ -327,6 +327,29 @@ export interface TappedFairGroup extends TappedEntityGroup {
 }
 
+/**
+ * A user taps a grouped card (Explore By, Discover Something new)
```
**suggestion**: "A user taps a grouping of cards (one of our supported home view section types)"? Maybe drop the specific "Explore By" / "Discover Something New" examples. Not sure if that's super meaningful out of context.

dblandin commented on src/Schema/Events/Tap.ts:
````diff
@@ -327,6 +327,29 @@ export interface TappedFairGroup extends TappedEntityGroup {
+ *  ```
+ *  {
+ *    action: "tappedCardGroup",
+ *    context_module: "exploreBySection" | discoverSomethingNewSection,
````
**suggestion**: I'd pick one of these strings for the example and drop the other as it becomes invalid JSON.

dblandin commented on src/Schema/Events/Tap.ts:
```diff
@@ -327,6 +327,29 @@ export interface TappedFairGroup extends TappedEntityGroup {
+ *    action: "tappedCardGroup",
+ *    context_module: "exploreBySection" | discoverSomethingNewSection,
+ *    context_screen_owner_type: "home",
+ *    destination_screen_owner_type: "collectionsCategory" | "collection",
```
**suggestion**: Same here, I'd pick one string to use in the example.

dblandin commented on src/Schema/Events/Tap.ts:
```diff
@@ -327,6 +327,29 @@ export interface TappedFairGroup extends TappedEntityGroup {
+ *    context_module: "exploreBySection" | discoverSomethingNewSection,
+ *    context_screen_owner_type: "home",
+ *    destination_screen_owner_type: "collectionsCategory" | "collection",
+ *    destination_screen_owner_id: "5359794d1a1e86c3740001f7" | undefined,
```
**suggestion**: Even though technically this can be undefined as the type allows, I'd specify just the string value here in the example.

dblandin commented on src/Schema/Events/Tap.ts:
```diff
@@ -327,6 +327,29 @@ export interface TappedFairGroup extends TappedEntityGroup {
+ *    destination_screen_owner_type: "collectionsCategory" | "collection",
+ *    destination_screen_owner_id: "5359794d1a1e86c3740001f7" | undefined,
+ *    destination_screen_owner_slug: "artworks-under-1000",
+ *    horizontal_slide_position: 1 | undefined,
```
**suggestion**: Same here. Even though technically this can be undefined as the type allows, I'd specify just the integer value here in the example.

dblandin reviewed (approved):
Awesome! Just a few minor suggestions for description/example in the code comment.

araujobarret commented:
Thanks for the detailed review Devon, I've addressed the last comments here https://github.com/artsy/cohesion/pull/536/commits/21f319ec0b4d5b6a16a630e06969714e2ca455fa and renamed the respective context modules to match what we agreed upon 👍 
once that's 🟢 and merged will update the MP PR

dblandin commented on src/Schema/Values/OwnerType.ts:
```diff
@@ -33,6 +33,7 @@ export enum OwnerType {
   collect = "collect",
   collection = "collection",
   collections = "collections",
+  collectionsCategory = "collectionsCategory",
```
This is very easy to miss (and annoying to discover later) but this value won't be easily accessible by Eigen unless it's also included in the ScreenOwnerType definition below ~L158.

araujobarret replied:
oh good catch, I think we can enhance these typings(ScreenOwnerType) in the future with Omit and Pick to make it lighter for engineers working with it

dblandin replied:
Yeah, for engineers that truly want to work around the typing they can always force it with `"somethingElse" as ScreenOwnerType` but at least that's made explicit in the downstream code.

artsyit commented:
<!-- GITHUB_RELEASE COMMENT: released -->
:rocket: PR was released in [`v4.213.0`](https://github.com/artsy/cohesion/releases/tag/v4.213.0) :rocket:
